tests/src/test/330/scala/com/nvidia/spark/rapids/IntervalDivisionSuite.scala

```diff
@@ -0,0 +1,232 @@
+        StructField("c_s", ShortType),
+        StructField("c_i", IntegerType),
+        StructField("c_l", LongType),
+        StructField("c_f", FloatType),
+        StructField("c_d", DoubleType)
+      ))
+      spark.createDataFrame(spark.sparkContext.parallelize(data), schema)
+    }) {
+    df =>
+      df.selectExpr(
+        "c_ym / c_b", "c_ym / c_s", "c_ym / c_i", "c_ym / c_l", "c_ym / c_f", "c_ym / c_d",
+        "c_ym / cast('15' as byte)", "c_ym / cast('15' as short)", "c_ym / 15",
+        "c_ym / 15L", "c_ym / 15.1f", "c_ym / 15.1d",
+        "interval '55' month / c_b",
+        "interval '15' month / c_s",
+        "interval '25' month / c_i",
+        "interval '15' month / c_l",
+        "interval '25' month / c_f",
+        "interval '35' month / c_d")
+  }
+
+  testSparkResultsAreEqual("test year-month interval / float num, normal case",
+    spark => {
+      val numRows = 1024
+      val r = new Random(0)
+
+      def getSign: Int = if (r.nextBoolean()) 1 else -1
+
+      val data = (0 until numRows).map(i => {
+        Row(Period.ofMonths(getSign * r.nextInt(Int.MaxValue)),
+          getSign * (r.nextFloat() + 10f), // add 10 to overflow
+          getSign * (r.nextDouble() + 10d))
+      })
+
+      val schema = StructType(Seq(
+        StructField("c_ym", YearMonthIntervalType()),
+        StructField("c_f", FloatType),
+        StructField("c_d", DoubleType)))
+      spark.createDataFrame(spark.sparkContext.parallelize(data), schema)
+    }
+  ) {
+    df => df.selectExpr("c_ym / c_f", "c_ym / c_d")
+  }
+
+  testSparkResultsAreEqual("test year-month interval / int num, normal case",
+    spark => {
+      val numRows = 1024
+      val r = new Random(0)
+
+      def getSign: Int = if (r.nextBoolean()) 1 else -1
+
+      val data = (0 until numRows).map(i => {
+        Row(Period.ofMonths(getSign * r.nextInt(Int.MaxValue)),
+          getSign * r.nextInt(1024 * 1024) + 1, // add 1 to avoid dividing 0
+          (getSign * r.nextInt(1024 * 1024) + 1).toLong)
+      })
+
+      val schema = StructType(Seq(
+        StructField("c_ym", YearMonthIntervalType()),
+        StructField("c_i", IntegerType),
+        StructField("c_l", LongType)))
+      spark.createDataFrame(spark.sparkContext.parallelize(data), schema)
+    }
+  ) {
+    df => df.selectExpr("c_ym / c_i", "c_ym / c_l")
+  }
+
+  // both gpu and cpu will throw ArithmeticException
+  def testDivideYMOverflow(testCaseName: String, months: Int, num: Any): Unit = {
+    testBothCpuGpuExpectedException[SparkException](testCaseName + ", cv / scalar",
+      e => e.getMessage.contains("ArithmeticException"),
+      spark => {
+        val data = Seq(Row(Period.ofMonths(months)))
+        val schema = StructType(Seq(
+          StructField("c1", YearMonthIntervalType())))
+        spark.createDataFrame(spark.sparkContext.parallelize(data), schema)
+      }
+    ) {
+      df =>
+        num match {
+          case Byte => df.selectExpr(s"c1 / cast('$num' as Byte)")
```

Comment:
This does not match properly, as it's checking an instance against a class which will never match.  It needs to be written something like this instead:
```suggestion
          case _: Byte => df.selectExpr(s"c1 / cast('$num' as Byte)")
```

Reply:
Sorry for this mistake. 
Updated.